asserts/asserts.go

```diff
@@ -1134,6 +1172,12 @@ func assembleAndSign(assertType *AssertionType, headers map[string]interface{},
 		return nil, err
 	}
 
+	if withJSONBody {
+		if err := checkJSON(assertType, body); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
same

Reply:
I can move this one though since the assertType is passed in

Reply:
I can move this one though since the assertType is passed in